src/main/scala/viper/gobra/frontend/Desugar.scala

```diff
@@ -3200,6 +3273,27 @@ object Desugar {
       f
     }
 
+    def pushFor(node: PNode, info: TypeInfo): String = {
+      val codeRoot = info.codeRoot(node)
+      val (stack, max) = continueCounter.getOrElse(codeRoot, (Seq(), 0))
+      continueCounter += (codeRoot -> (Seq(max + 1) ++ stack, max + 1))
+      CONTINUE_LABEL_PREFIX + (max + 1)
+    }
+
+    /** returns a unique identifier for a for stmt*/
+    def fetchForId(node: PNode, up: Int, info: TypeInfo): String = {
```

Comment:
why do we use the id here? Wouldn't it be enough ti just pass the node and then find it in the stack? Is it some kind of optimization?

Reply:
You are right, we can use the info object to produce the 'up' argument inside instead of computing it outside.